test/thumbnailer_test.cc

```diff
@@ -31,47 +32,42 @@ void WebPDataDelete(WebPData* webp_data) {
   delete webp_data;
 }
 
-class WebPSamplesGenerator {
+class WebPTestGenerator {
  public:
   // Initializing.
-  WebPSamplesGenerator()
+  WebPTestGenerator()
       : pic_count_(10),
         width_(kDefaultWidth),
         height_(kDefaultHeight),
         transparency_(0xff),
         randomized_(true) {}
 
-  WebPSamplesGenerator(int pic_count, uint8_t transparency, bool randomized)
+  WebPTestGenerator(int pic_count, uint8_t transparency, bool randomized)
       : pic_count_(pic_count),
         width_(kDefaultWidth),
         height_(kDefaultHeight),
         transparency_(transparency),
         randomized_(randomized) {}
 
-  WebPSamplesGenerator(int pic_count, int width, int height,
-                       uint8_t transparency, bool randomized)
+  WebPTestGenerator(int pic_count, int width, int height, uint8_t transparency,
+                    bool randomized)
       : pic_count_(pic_count),
         width_(width),
         height_(height),
         transparency_(transparency),
         randomized_(randomized) {}
 
-  // Returns vector of WebPPicture(s).
-  // randomized == true: picture is random noise
-  // randomized == false: picture is random solid color
-  std::vector<std::unique_ptr<WebPPicture, void (*)(WebPPicture*)>>
-  GeneratePics() {
-    std::vector<std::unique_ptr<WebPPicture, void (*)(WebPPicture*)>> pics;
+  // Returns true on success and false on failure.
+  bool GenerateWebPData(WebPData* const webp_data) {
+    libwebp::Thumbnailer thumbnailer = libwebp::Thumbnailer();
+    auto test_pics = GeneratePics();
     for (int i = 0; i < pic_count_; ++i) {
-      pics.emplace_back(new WebPPicture, WebPPictureDelete);
-      auto& pic = pics.back();
-      WebPPictureInit(pic.get());
-      pic->use_argb = 1;
-      pic->width = width_;
-      pic->height = height_;
-      WebPPictureImportRGBA(pic.get(), GenerateRGBA(i).data(), width_ * 4);
+      if (thumbnailer.AddFrame(*test_pics[i].get(), i * 500) !=
```

Comment:
For `unique_ptr` the `*` operator is overridden so that you do not need the `.get()`